xayn-ai/src/coi/relevance.rs

```diff
@@ -169,6 +171,10 @@ impl RelevanceMap {
     pub(super) fn clean(&mut self, coi_id: CoiId, relevance: Relevance, key_phrase: &KeyPhrase) {
         if let Some(key_phrases) = self.relevance_to_key_phrase.get_mut(&(relevance, coi_id)) {
             key_phrases.retain(|this| this != key_phrase);
+            self.cleaned
+                .entry(coi_id)
+                .or_default()
+                .push((relevance, key_phrase.clone()));
```

Comment:
it seems wasteful to clone the key phrase which we have just removed. instead of `key_phrases.retain()` we could find the index of the `key_phrase` in the `key_phrases` via `Iterator::position()` and `Vec::remove()` it and then `push()` it to the `cleaned` map.

this will also solve the remaining issue that even though the pushing is now inside the `if let`, it is still possible that we push a key phrase to `cleaned` which never existed in `key_phrases` before (the retaining will not remove anything if the `key_phrase` doesn't exist).

overall, we probably need more tests for the `RelevanceMap` methods to cover all those edge cases, maybe in a follow up pr.